### ISSUE TITLE
Transaction not updating when mined / Network delay

### ISSUE THREAD

**Issue by KingOfAllChunks** [MetaMask/metamask-plugin#1157]
Hello. I've been delveloping a Dapp using your wonderful Chrome plugin. However, I have noticed that sometimes, when my Dapp sends a transaction to a contract, the transaction is displayed as "pending" on the UI of the extension, and my frontend web3 is not receiving any events from the contract that have been fired with the transaction. 

However, the transaction is successfully mined if we check on the blockchain explorer.

It can take from a few minutes, even to half an hour for the transaction to update and for my application to receive the events.

I'm currently on Ropsten test network with a solid internet connection.
Sending multiple transactions does not make the previous ones update faster.

It seems like the issue seems like lack of connection to the backend, or a problem delaying connection to the blockchain in general.

My Dapp can extract older events fired from the contract from before, or execute constant contract methods, but everything else works with a huge delay.

Do you have an idea what's the problem.

**Comment by flyswatter** [comment]
One theory is that this could be because the ropsten network has been undergoing an attack.

For the moment, I'd suggest using your own test network and main net primarily.

https://www.reddit.com/r/ethereum/comments/5w3vqu/ropsten_spam/?st=IZR8X5CW&sh=9bbb8735

**Comment by KingOfAllChunks** [comment]
So you're saying that this is not a Metamask issue? If so, then why does blockchain explorer update the transaction status properly, and the extension is still showing it as pending (1 hour and 7 minutes as of initial transaction initiation, still growing)?

I have a local test network, but with that I cannot develop / test contracts unless I'm on my primary computer. Plus the time for transaction mining is instant, which is not sufficient if I want to test real-world scenarios.

Is there, by chance, a different stable-ish test network, or I'll have to host my own and make it public?

**Comment by flyswatter** [comment]
I'm not sure if it is, it's a theory.

For your testing concerns:
`testrpc` does have a mining delay flag.
The main-net is another stable network.
Parity also lets you deploy custom networks freely.

In terms of debugging, it might help if you could give us a UI state dump:
https://github.com/MetaMask/faq/blob/master/LOGS.md#popup-logs-chrome

**Comment by KingOfAllChunks** [comment]
Here's the log: http://pastebin.com/WCyuUvCg
My transactions were updated after an hour and 10-20 minutes. I've issued a new one, which as of now (5 minutes from issuing) still is not updated. 

The main-net is nice, but I'd like to test without huge additional costs (like intensive contract publishing).

Is there a way to preserve testrpc's blockchain, so that after a restart it still keeps the contracts inside of it? And can I publish it somehow for it to be accessed from outside? I'm guessing that simple port forwarding is enough, right? Sorry, I don't have experience with Pairity.
